SU2_CFD/src/solver_direct_heat.cpp

```diff
@@ -578,3 +578,1978 @@ void CHeatSolver::ImplicitEuler_Iteration(CGeometry *geometry, CSolver **solver_
+
+            Normal = geometry->vertex[iMarker][iVertex]->GetNormal();
+            Area = 0.0;
+            for (iDim = 0; iDim < nDim; iDim++) Area += Normal[iDim]*Normal[iDim];
+            Area = sqrt (Area);
+
+            HeatFluxDensity     = GetConjugateHeatVariable(iMarker, iVertex, 1);
+            HeatFluxDensity     = HeatFluxDensity/(rho_cp_solid*Temperature_Ref);
+
+            HeatFluxValue       = HeatFluxDensity * Area;
+
+            Res_Visc[0] = -HeatFluxValue;
+            LinSysRes.SubtractBlock(iPoint, Res_Visc);
+          }
+        }
+      }
+    }
+  }
+}
+
+void CHeatSolverFVM::Heat_Fluxes(CGeometry *geometry, CSolver **solver_container, CConfig *config) {
+
+  unsigned long iVertex, iPoint, iPointNormal;
+  unsigned short Boundary, Monitoring, iMarker, iMarker_HeatFlux, iDim;
+  su2double *Coord, *Coord_Normal, *Normal, Area, dist, Twall, dTdn, cp_fluid, rho_cp_solid,
+      eddy_viscosity, thermal_conductivity, thermal_diffusivity;
+  string Marker_Tag, HeatFlux_Tag;
+  bool flow = ((config->GetKind_Solver() == NAVIER_STOKES)
+               || (config->GetKind_Solver() == RANS)
+               || (config->GetKind_Solver() == DISC_ADJ_NAVIER_STOKES)
+               || (config->GetKind_Solver() == DISC_ADJ_RANS));
+
+#ifdef HAVE_MPI
+  su2double MyAllBound_HeatFlux, MyAllBound_AvgTemperature;
+#endif
+
+  cp_fluid = config->GetSpecificHeat_Fluid();
+  rho_cp_solid = config->GetSpecificHeat_Solid()*config->GetDensity_Solid();
+
+  AllBound_HeatFlux = 0.0;
+  AllBound_AvgTemperature = 0.0;
+
+  for ( iMarker = 0; iMarker < nMarker; iMarker++ ) {
+
+    AvgTemperature[iMarker] = 0.0;
+
+    Boundary = config->GetMarker_All_KindBC(iMarker);
+    Marker_Tag = config->GetMarker_All_TagBound(iMarker);
+    Monitoring = config->GetMarker_All_Monitoring(iMarker);
+
+    Heat_Flux[iMarker] = 0.0;
+
+    if ( Boundary == ISOTHERMAL && Monitoring == YES) {
+
+      Twall = config->GetIsothermal_Temperature(Marker_Tag)/config->GetTemperature_Ref();
+
+      for( iVertex = 0; iVertex < geometry->nVertex[iMarker]; iVertex++ ) {
+
+        iPoint = geometry->vertex[iMarker][iVertex]->GetNode();
+
+        if(geometry->node[iPoint]->GetDomain()) {
+
+          iPointNormal = geometry->vertex[iMarker][iVertex]->GetNormal_Neighbor();
+
+          Coord = geometry->node[iPoint]->GetCoord();
+          Coord_Normal = geometry->node[iPointNormal]->GetCoord();
+
+          Normal = geometry->vertex[iMarker][iVertex]->GetNormal();
+          Area = 0.0;
+          for (iDim = 0; iDim < nDim; iDim++) Area += Normal[iDim]*Normal[iDim]; Area = sqrt(Area);
+
+          dist = 0.0;
+          for (iDim = 0; iDim < nDim; iDim++) dist += (Coord_Normal[iDim]-Coord[iDim])*(Coord_Normal[iDim]-Coord[iDim]);
+          dist = sqrt(dist);
+
+          dTdn = (Twall - node[iPointNormal]->GetSolution(0))/dist;
+
+          if(flow) {
+            // eddy viscosity should be added?
```

Comment:
On the wall, eddy viscosity should be zero. In the volume, it should be included to get the effective conductivity (unless this is set constant by the user).

Reply:
Agreed. Will remove these questions!